AK/JsonValue.h

```diff
@@ -91,6 +91,22 @@ class JsonValue {
     u32 to_u32(u32 default_value = 0) const { return to_number<u32>(default_value); }
     u64 to_u64(u64 default_value = 0) const { return to_number<u64>(default_value); }
 
+    String to_bcd(u32 default_value = 0) const
+    {
+        StringBuilder bcd_result;
+        u32 as_u32 = to_number<u32>(default_value);
+
+        while (as_u32 != 0) {
+            u8 nybble = (as_u32 & 0xf);
+            bcd_result.append(nybble + '0');
```

Comment:
> (also IMO this should be a free function in lsusb.cpp)

Agreed, I'll move this into `lsusb.cpp`

As for each `nybble` being between 0 and 9, they should be unless the manufacturer has busted their BCD value burned into the USB device, though obviously relying on this isn't too optimal.

Reply:
Was wondering because on my system I have BCD values like `3.10` and `5.13`, though I could be reading this all wrong:
```
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               3.10
  bDeviceClass            9 Hub
  bDeviceSubClass         0 
  bDeviceProtocol         3 
  bMaxPacketSize0         9
  idVendor           0x1d6b Linux Foundation
  idProduct          0x0003 3.0 root hub
  bcdDevice            5.13
  iManufacturer           3 Linux 5.13.0-23-generic xhci-hcd
  iProduct                2 xHCI Host Controller
  iSerial                 1 0000:01:00.2
  bNumConfigurations      1
```

Reply:
So eathat makes sense; each hex nybble is from 0 to 9. Why the decimal point is there, I'm not sure. I'll have to go over the spec again. Perhaps the first two nybbles are xx and the rest are decimal points

Reply:
Gotcha, so `3.10` is `0x310`, not `0x3a`? Yeah that decimal placement is weird then

Reply:
> Gotcha, so `3.10` is `0x310`, not `0x3a`? Yeah that decimal placement is weird then

Yeah that's it.

Also the spec says this:

JJ.M.N (0xJJMN)

Where J is Major version
M is Minor Version
N is sub Minor

There was an argument about the validity of thie statement in the spec here: https://linux-usb-devel.narkive.com/f1vvC3aQ/bcd-version-numbers-in-proc-bus-usb-devices